resources/js/oe_whitelabel.project_status.js

```diff
@@ -0,0 +1,135 @@
+   *   The parameter is a value within the [0..nTicks] interval.
+   */
+  function animateProgress(msBegin, msEnd, nTicks, setProgress) {
+    const msNow = Date.now();
+    const msTick = (msEnd - msBegin) / nTicks;
+    const tickNext = Math.ceil((msNow - msBegin) / msTick);
+
+    if (tickNext >= nTicks) {
+      setProgress(nTicks);
+      return;
+    }
+
+    // Compute a delay for the start of the interval.
+    const msIntervalDelay = msBegin - msNow + Math.max(0, tickNext) * msTick;
+
+    // Register a repeated interval.
+    window.setTimeout(function () {
+      let tick = Math.max(0, tickNext);
+      setProgress(tick);
+      if (tick >= nTicks) {
+        return;
+      }
+      ++tick;
+      const intervalId = window.setInterval(function () {
+        setProgress(tick);
+        if (tick >= nTicks) {
+          clearInterval(intervalId);
+        }
+        ++tick;
+      }, msTick);
+    }, msIntervalDelay);
+
+    // Set initial progress.
+    setProgress(Math.max(0, tickNext - 1));
+  }
+
+  /**
+   * Animates the project status badge and progress bar.
+   *
+   * @type {Drupal~behavior}
+   *
+   * @prop {Drupal~behaviorAttach} attach
+   *   Initialises the behavior.
+   */
+  Drupal.behaviors.projectStatus = {
+    attach: function (context) {
+      $('.bcl-project-status', context).once('bcl-project-status').each(function () {
+        const $element = $(this);
+        let msBegin = $element.data('start-timestamp') * 1000;
+        let msEnd = $element.data('end-timestamp') * 1000;
+        const statusLabels = $element.data('status-labels').split('|');
+
+        // Process the status label.
+        $('.badge', this).each(function () {
+          const $element = $(this);
+          animateStatus(msBegin, msEnd, function (status) {
+            $element.removeClass(colorClasses);
+            $element.addClass(colorClasses[status]);
+            if (statusLabels) {
+              $element.html(statusLabels[status]);
+            }
+          });
+        });
+
+        // Process the progress bar.
+        $('.progress-bar', this).each(function () {
+          const $element = $(this);
+          animateStatus(msBegin, msEnd, function (status) {
+            $element.removeClass(colorClasses);
+            $element.addClass(colorClasses[status]);
+          });
+          // Disable css transition.
+          // It looks bad for the initial setting, and afterwards it does not help.
+          $element.css('transition', false);
+          const factor = 2;
+          animateProgress(msBegin, msEnd, factor * 100, function (tick) {
+            const percent = tick / factor;
+            $element.css('width', percent + '%');
+            $element.attr('aria-valuenow', percent);
+            $element.attr('aria-label', percent);
+          });
+        });
+
+        // Reveal the entire section.
+        $element.removeClass('d-none');
+      });
+    }
+  };
+
+})(bootstrap, Drupal, jQuery);
```

Comment:
There seems to be a limit for window.setTimeout() due to 32 bit integer. If the timeout duration is higher than that number, it will happen immediately instead. The highest duration seems to be around 10 years.

Try in browser console: `window.setTimeout(console.log, 10 * 360 * 24 * 60 * 60 * 1000, '10 years passed'); window.setTimeout(console.log, 11 * 360 * 24 * 60 * 60 * 1000, '11 years passed');`.

See https://stackoverflow.com/questions/3468607/why-does-settimeout-break-for-large-millisecond-delay-values

Either way I will remove this stuff, too much risk and it was not asked for.